src/internal/operators/withLatestFrom.ts

```diff
@@ -66,88 +147,68 @@ export function withLatestFrom<T, R>(array: ObservableInput<any>[], project: (..
  * each input Observable.
  * @name withLatestFrom
  */
-export function withLatestFrom<T, R>(...args: Array<ObservableInput<any> | ((...values: Array<any>) => R)>): OperatorFunction<T, R> {
+export function withLatestFrom<T, R>(...inputs: any[]): OperatorFunction<T, R> {
   return (source: Observable<T>) => {
-    let project: any;
-    if (typeof args[args.length - 1] === 'function') {
-      project = args.pop();
+    let project: (...values: any[]) => R;
+    if (typeof inputs[inputs.length - 1] === 'function') {
+      project = inputs.pop();
     }
-    const observables = <Observable<any>[]>args;
-    return lift(source, new WithLatestFromOperator(observables, project));
-  };
-}
-
-class WithLatestFromOperator<T, R> implements Operator<T, R> {
-  constructor(private observables: Observable<any>[],
-              private project?: (...values: any[]) => Observable<R>) {
-  }
-
-  call(subscriber: Subscriber<R>, source: any): any {
-    return source.subscribe(new WithLatestFromSubscriber(subscriber, this.observables, this.project));
-  }
-}
 
-/**
- * We need this JSDoc comment for affecting ESDoc.
- * @ignore
- * @extends {Ignored}
- */
-class WithLatestFromSubscriber<T, R> extends ComplexOuterSubscriber<T, R> {
-  private values: any[];
-  private toRespond: number[] = [];
+    return lift(source, function (this: Subscriber<R | T[]>, source: Observable<T>) {
+      const subscriber = this;
+      const len = inputs.length;
+      const otherValues = new Array(len);
+      // An array of whether or not the other sources have emitted. Matched with them by index.
+      // TODO: At somepoint, we should investigate the performance implications here, and look
+      // into using a `Set()` and checking the `size` to see if we're ready.
+      let hasValue: Record<string, any> = inputs.map(() => false);
```

Comment:
issue: IMO, this should be typed as `boolean[]` 'cause that more accurately represents what it is. The fact that it's typed as `Record<string, any>` - with emphasis on the **any** is why `every` can be called on it, despite `every` not being a `Record` method.

Reply:
Removed the incorrect typing to let inference do it's thing